app/src/main/java/com/otaman/notes/model/room/NoteDatabase.kt

```diff
@@ -13,16 +13,16 @@ abstract class NoteDatabase: RoomDatabase() {
     companion object {
         @Volatile
         private var INSTANCE: NoteDatabase? = null
-        fun getDatabase(context: Context): NoteDatabase {
-            return INSTANCE ?: synchronized(this) {
-                val instance = Room.databaseBuilder(
-                    context.applicationContext,
-                    NoteDatabase::class.java,
-                    "note_database"
-                ).build()
-                INSTANCE = instance
-                instance
+        fun getInstance(context: Context): NoteDatabase =
+            INSTANCE ?: synchronized(this) {
+                INSTANCE ?: buildDatabase(context).also { INSTANCE = it }

```

Comment:
`INSTANCE` is always null on line 18. Actually, I don't see any issues with `NoteDatabase` class before changes